llvm/utils/lit/lit/llvm/config.py

```diff
@@ -632,15 +632,28 @@ def use_clang(
             self.add_tool_substitutions(tool_substitutions)
             self.config.substitutions.append(("%resource_dir", builtin_include_dir))
 
-        self.config.substitutions.append(
-            (
-                "%itanium_abi_triple",
-                self.make_itanium_abi_triple(self.config.target_triple),
+        # There will be no default target triple if one was not specifically
+        # set, and the host's architecture is not an enabled target.
+        if (
+            self.config.target_triple
+            and self.config.target_triple != "@LLVM_TARGET_TRIPLE@"
```

Comment:
What if both default target triple and LLVM_TARGET_TRIPLE are set to the same target?

Reply:
My understanding is that `LLVM_TARGET_TRIPLE` is the default target.

First reference to it is `<build dir>/lib/cmake/llvm/LLVMConfig.cmake`:
```
set(LLVM_TARGET_TRIPLE "")

set(LLVM_HOST_TRIPLE "aarch64-unknown-linux-gnu")
```
For a clang that reports:
```
$ ./bin/clang --version
clang version 18.0.0git (https://github.com/llvm/llvm-project.git dd71751f6e0e0ece16184a7428ecd160ef320bc6)
Target: unknown
Thread model: posix
InstalledDir: /home/david.spickett/build-llvm-no-host/./bin
```
And the majority of lit configs are using:
```
config.target_triple = "@LLVM_TARGET_TRIPLE@"
```
So if you have a default target, "@LLVM_TARGET_TRIPLE@" expands to that triple, and `config.target_triple` will have that same value.

The `self.config.target_triple != "@LLVM_TARGET_TRIPLE@"` is I think to catch when the `@...@` isn't expanded at all. Not sure how that would happen, just following the `host_triple` logic there.

Reply:
But then why do we need the check since we only use self.config.target_triple below? Why not only check for self.config.target_triple?

Reply:
I'm not sure what you mean, please suggest what you think the code should look like.

Reply:
Since we only use self.config.target_triple inside the if I don't understand why we need to test whether @LLVM_TARGET_TRIPLE@ has been correctly substituted. In other words, why not simply:

```suggestion
        if self.config.target_triple:
```

I presume self.config.target_triple will be set to @LLVM_TARGET_TRIPLE@ elsewhere when there is no default target but the variable is set.

Reply:
Sure, the code below for `config.host_triple` does both but I can't explain exactly when it would not be expanded. Easy enough to add the same check for `config.target_triple` if anyone does manage to hit that situation in future.

I've applied your suggestion.